django/core/checks/urls.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import unicode_literals
+
+from . import Tags, Warning, register
+
+
+@register(Tags.urls)
+def check_url_config(app_configs, **kwargs):
+    from django.core.urlresolvers import get_resolver
+    resolver = get_resolver()
+    return check_resolver(resolver)
+
+
+def check_resolver(resolver):
+    """
+    Recursively check the resolver
+    """
+    from django.core.urlresolvers import RegexURLPattern, RegexURLResolver
+    warnings = []
+    for pattern in resolver.url_patterns:
+        if isinstance(pattern, RegexURLResolver):
+            warnings.extend(check_include_trailing_dollar(pattern))
+            # Check resolver recursively
+            warnings.extend(check_resolver(pattern))
+        elif isinstance(pattern, RegexURLPattern):
+            warnings.extend(check_pattern_name(pattern))
+
+        warnings.extend(check_pattern_startswith_slash(pattern))
+
+    return warnings
+
+
+def describe_pattern(pattern):
+    """
+    Formats the URL pattern for display in warning messages
```

Comment:
I vaguely remembered a discussion about doc strings, so I checked the style guide, but still got it wrong because the advice changed between 1.8 and dev!  Sorry about the punctuation.